src/build/helpers.ts

```diff
@@ -325,3 +325,21 @@ export function followTypeReferences(
     }
   }
 }
+
+export function assertUnique(list: string[]): string[] {
+  if (!Array.isArray(list)) return list;
+  const result = new Set();
+
+  list
+    .map((str) => str.trim())
+    .forEach((str, _, trimmed) => {
+      if (trimmed.indexOf(str) !== trimmed.lastIndexOf(str)) result.add(str);
+    });
+
+  if (result.size) {
+    throw new Error(
+      `Duplicate items [${[...result].join(", ")}] found in the list.`
+    );
+  }
+  return list;
```

Comment:
Good but IMO a bit overkill, checking whether it has a duplicate is enough IMO. A human should easily see the duplications in the list, given that callers are only expected to pass small lists.

```suggestion
  if (new Set(list).size < list.length) {
    throw new Error(`Duplicate items found in the list: ${list}`);
  }
  return list;
```